.github/workflows/main.yml

```diff
@@ -0,0 +1,17 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+    types: [opened, synchronize]
```

Comment:
I don't see the point of specifying this. The default pull request events should be fine.